azurelinuxagent/common/utils/restutil.py

```diff
@@ -553,10 +563,12 @@ def http_put(url,
 def http_delete(url,
                 headers=None,
                 use_proxy=False,
-                max_retry=DEFAULT_RETRIES,
+                max_retry=None,
                 retry_codes=None,
                 retry_delay=DELAY_IN_SECONDS):
 
+    if max_retry is None:
+        max_retry = DEFAULT_RETRIES
```

Comment:
Why not just make this check part of the `http_request` function? Would avoid having this `None` check for every `http_*` function calls separately

Reply:
Each of these function had its own default (which happened to be the same). I did not want to break that interface.